volatility3/framework/plugins/windows/processghosting.py

```diff
@@ -33,52 +35,163 @@ def get_requirements(cls):
             ),
         ]
 
-    def _generator(self, procs):
-        kernel = self.context.modules[self.config["kernel"]]
+    @classmethod
+    def _process_checks(
+        cls,
+        proc: interfaces.objects.ObjectInterface,
+        mapped_files: Dict[int, Tuple[str, interfaces.objects.ObjectInterface]],
+    ) -> Generator[
+        Tuple[int, Optional[int], Optional[int], int, Optional[str]], None, None
+    ]:
+        """
+        Checks the EPROCESS for signs of ghosting
+        """
+        if not proc.has_member("ImageFilePointer"):
+            return
 
-        if not kernel.get_type("_EPROCESS").has_member("ImageFilePointer"):
-            vollog.warning(
-                "This plugin only supports Windows 10 builds when the ImageFilePointer member of _EPROCESS is present"
+        delete_pending = None
+
+        # if it is 0 then its a side effect of process ghosting
+        if proc.ImageFilePointer.vol.offset != 0:
+            try:
+                file_object = proc.ImageFilePointer
+                delete_pending = file_object.DeletePending
+                file_object = file_object.dereference().vol.offset
+            except exceptions.InvalidAddressException:
+                file_object = 0
+
+        # ImageFilePointer equal to 0 means process ghosting or similar techniques were used
+        else:
+            file_object = 0
+
+        # delete_pending besides 0 or 1 = smear
+        if isinstance(delete_pending, int) and delete_pending not in [0, 1]:
+            vollog.debug(
```

Comment:
Just a debug message?  Feels a bit strange if it's not an `int` (that's kind of on us, unless the typing changes upstream by MS, in which case we probably want to know)?

Reply:
We set it to None on line 52 as the process might not have the member. So then we check if its an int (was set in the ImageFilePointer path). Just a debug message as its smeared out if its anything but 0 or 1.